timechart/data_io/import_strategies/settings_import_strategy.py

```diff
@@ -0,0 +1,159 @@
+# Base Import Strategy Class
+
+from abc import ABC, abstractmethod
+from qtpy.QtGui import QColor
+
+from ...displays.defaults import ASYNC_DATA_SAMPLING, SYNC_DATA_SAMPLING
+from ...utilities.utils import random_color
+
+
+class SettingsImportException(Exception):
+    pass
+
+
+class SettingsImportStrategy:
+    __metaclass__ = ABC
+
+    def __init__(self, pydm_main_dislay):
+        self.main_display = pydm_main_dislay
+
+    @abstractmethod
+    def import_file(self, settings_file):
+        raise NotImplementedError
+
+    @classmethod
+    def apply_settings(cls, settings, main_display):
```

Comment:
apply_settings() runs the common code needed by TimeChartImportStrategy and StripToolImportStrategy. Its purpose is to avoid duplicate code. Where else do you see the same code being repeated?

Reply:
It is about 130 lines of setting widgets up and firing their signals. There should be a way to break this up and avoid copying code with the widget name switched.

Reply:
I consolidated the widget settings in the latest commit.